workflow/artifacts/artifactory/artifactory.go

```diff
@@ -0,0 +1,66 @@
+package artifactory
+
+import (
+	wfv1 "github.com/argoproj/argo/api/workflow/v1alpha1"
+	"net/http"
+	"os"
+	"github.com/argoproj/argo/errors"
+	"io"
+)
+
+type ArtifactoryArtifactDriver struct {
+	Username string
+	Password string
+}
+
+
+// Download artifact from an artifactory URL
+func (a *ArtifactoryArtifactDriver) Load(artifact *wfv1.Artifact, path string) error {
+
+	lf, err := os.Create(path)
+	if err != nil {
+		return err
+	}
+	defer lf.Close()
+
+	req,err := http.NewRequest(http.MethodGet,artifact.Artifactory.URL,nil)
+	if err != nil {
+		return err
+	}
+	req.SetBasicAuth(a.Username,a.Password)
+	res, err := (&http.Client{}).Do(req)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+	if res.StatusCode < 200 || res.StatusCode >= 300 {
+		return errors.InternalErrorf("loading file from artifactory failed with reason:%s",res.Status)
```

Comment:
Do you want to log the status as well as the reason? Currently, only the status is being logged.

Reply:
status includes the http status code and reason string